proposals/extend_cli_for_generate_policies.md

```diff
@@ -0,0 +1,150 @@
+	Policy            string              `json:"policy"`
+	Rule              string              `json:"rule"`
+	Result            report.PolicyResult `json:"result"`
+	Status            report.PolicyResult `json:"status"`
+	Resource          string              `json:"resource"`
+	Kind              string              `json:"kind"`
+	Namespace         string              `json:"namespace"`
+	PatchedResource   string              `json:"patchedResource"`
+	AutoGeneratedRule string              `json:"auto_generated_rule"`
+	GeneratedResource string              `json:"generatedResource"`
+	CloneResource     string              `json:"cloneResource"`
+}
+```
+
+### Method to test data object
+
+The method is similar to testing mutate policies. User will provide the desired resource manifest file as generatedResource in the results array of the test file. This manifest file will be compared with the Kyverno generated resource and accordingly the test will display the result as pass or fail.
+
+Additionally, like in the case of testing mutate policies, we provide the resource on which we want to perform mutation. For generate policies, the triggering resource should be provided on creation/update of which our generation rule is applied.
+
+#### Example: Add Network Policy
+
+A sample test file for [this policy](https://kyverno.io/policies/best-practices/add_network_policy/) will look like this -
+
+```yaml
+name: deny-all-traffic
+policies:
+  - policy.yaml
+resources:
+  - resource.yaml
+results:
+  - policy: add-networkpolicy
+    rule: default-deny
+    resource: hello-world-namespace
+    generatedResource: generatedResource.yaml
+    kind: Namespace
+    result: pass
+```
+resource.yaml is the manifest for creating a new Namespace resource. The result will pass if the generatedResource.yaml looks like this -
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny
+  namespace: hello-world-namespace
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+### Method to test clone object
+
+Generate rules with clone object requires a pre-existing resource in the cluster. The manifest of the rule will not have the definition of this resource. Instead, the rule will only specify the name of the resource and the namespace where it resides. To have the resource exist on the cluster before the rule is applied, user should provide the definition of the resource. The rest of the approach remains same as that of data object.
+
+#### Example: Sync Secrets
+
+Consider [this policy](https://kyverno.io/policies/other/sync_secrets/). The test file should look like this -
+
+```yaml
+name: sync-secrets
+policies:
+  - policy.yaml
+resources:
+  - resource.yaml
+results:
+  - policy: sync-secrets
+    rule: sync-image-pull-secret
+    resource: hello-world-namespace
+    generatedResource: generatedResource.yaml
+    cloneResource: cloneResource.yaml # Pre-existing resource definition
```

Comment:
I'd recommend calling this either "sourceResource" or "cloneSourceResource". Something that acknowledges this is the source of the clone.